### ISSUE TITLE
'allowed_redirect_uris' must be set if 'role_type' is 'oidc' or unspecified.

### ISSUE THREAD

**Issue by Rotwang** [terraform-providers/terraform-provider-vault#369]
Hi

### Terraform Version
Terraform v0.11.13
+ provider.vault v1.6.0

### Affected Resource(s)
vault_jwt_auth_backend_role

### Terraform Configuration Files
```hcl
resource "vault_jwt_auth_backend" "jwt" {
  path               = "jwt"
  oidc_discovery_url = "${var.tenant_url}"
  bound_issuer       = "${var.tenant_url}"
}

resource "vault_jwt_auth_backend_role" "myrole" {
  role_name       = "myrole"
  policies        = ["default"]
  bound_audiences = ["${var.app_id}"]
  user_claim      = "email"
  groups_claim    = "https://vault/groups"
  backend         = "${vault_jwt_auth_backend.jwt.path}"
}

```

### Actual Behavior
What actually happened?

```
Error: Error applying plan:

1 error(s) occurred:

* vault_jwt_auth_backend_role.myrole: 1 error(s) occurred:

* vault_jwt_auth_backend_role.myrole: error writing JWT auth backend role "auth/jwt/role/myrole": Error making API request.

URL: PUT https://myvault.com/v1/auth/jwt/role/myrole
Code: 400. Errors:

* 'allowed_redirect_uris' must be set if 'role_type' is 'oidc' or unspecified.
```

**Comment by jleechp-occm** [comment]
This error occurs with Vault 1.1.0 because of the new OIDC workflows.

**Comment by sjoby** [comment]
Confrim this is still an issue with Vault Provider version 1.7.0 and Vault Version 1.1.2.  "vault_jwt_auth_backend_role" need to be able to set "allowed_redirect_uris" as type "list".

**Comment by roidelapluie** [comment]
Please see #398

**Comment by kalafut** [comment]
Fixed in 1.8.0 by #398
